packages/framework/src/Model/Product/Search/Export/ProductSearchExportListener.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Model\Product\Search\Export;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+
+class ProductSearchExportListener
+{
+    /**
+     * @var \Doctrine\ORM\EntityManagerInterface
+     */
+    protected $entityManager;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Product\Search\Export\ProductSearchExportFacade
+     */
+    protected $productSearchExportFacade;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Product\Search\Export\ProductSearchExportScheduler
+     */
+    protected $productSearchExportScheduler;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Search\Export\ProductSearchExportScheduler $productSearchExportScheduler
+     * @param \Shopsys\FrameworkBundle\Model\Product\Search\Export\ProductSearchExportFacade $productSearchExportFacade
+     * @param \Doctrine\ORM\EntityManagerInterface $entityManager
+     */
+    public function __construct(
+        ProductSearchExportScheduler $productSearchExportScheduler,
+        ProductSearchExportFacade $productSearchExportFacade,
+        EntityManagerInterface $entityManager
+    ) {
+        $this->productSearchExportScheduler = $productSearchExportScheduler;
+        $this->productSearchExportFacade = $productSearchExportFacade;
+        $this->entityManager = $entityManager;
+    }
+
+    /**
+     * @param \Symfony\Component\HttpKernel\Event\FilterResponseEvent $filterResponseEvent
+     */
+    public function onKernelResponse(FilterResponseEvent $filterResponseEvent): void
+    {
+        if ($this->productSearchExportScheduler->hasAnyProductIdsForImmediateExport()) {
```

Comment:
Extract body of this method to another public method, so it can be called for example from imports